source/Patches/CustomHats/HatLoader.cs

```diff
@@ -47,7 +47,10 @@ internal static void LoadHats()
         private static HatMetadataJson LoadJson()
         {
             var stream = Assembly.GetManifestResourceStream($"{HAT_RESOURCE_NAMESPACE}.{HAT_METADATA_JSON}");
-            return JsonConvert.DeserializeObject<HatMetadataJson>(Encoding.UTF8.GetString(stream.ReadFully()));
+            
+            // The only JsonConvert.DeserializeObject generic that is available in Among Us
+            return JsonConvert.DeserializeObject<HatMetadataJson>(Encoding.UTF8.GetString(stream.ReadFully()), 
+                new JsonSerializerSettings { TypeNameHandling = TypeNameHandling.Auto });
```

Comment:
This is only called once, right?

Reply:
That's correct; there is static boolean flag that checks if it was executed [here](https://github.com/Anusien/Town-Of-Us/blob/mainline/source/Patches/CustomHats/Patches/AmongUsClient_Patches.cs#L8)